File: src/2424. Longest Uploaded Prefix.cpp
class LUPrefix {
private:
    int length = 0;
    unordered_set<int> nums;
public:
    LUPrefix(int n) {
    }
    
    void upload(int video) {
        nums.insert(video);
    }
    
    int longest() {
        while (nums.count(length + 1)) {
            length++;
        }
        return length;
    }
};

/**
 * Your LUPrefix object will be instantiated and called as such:
 * LUPrefix* obj = new LUPrefix(n);
 * obj->upload(video);
 * int param_2 = obj->longest();
 */

File: src/2424. Longest Uploaded Prefix.py
class LUPrefix:

    def __init__(self, n: int):
        self.length = 0
        self.nums = set()

    def upload(self, video: int) -> None:
        self.nums.add(video)

    def longest(self) -> int:
        while self.length + 1 in self.nums:
            self.length += 1
        return self.length


# Your LUPrefix object will be instantiated and called as such:
# obj = LUPrefix(n)
# obj.upload(video)
# param_2 = obj.longest()